keyboards/darkproject/kd87b_ltd/keymaps/via/keymap.c

```diff
@@ -0,0 +1,88 @@
+  KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,       KC_MINS,  KC_EQL,   KC_BSPC,  KC_INS,   KC_HOME,  KC_PGUP,  
+  KC_TAB,   KC_Q,     KC_W,     KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,       KC_LBRC,  KC_RBRC,  KC_BSLS,  KC_DEL,   KC_END,   KC_PGDN,    
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,    KC_QUOT,            KC_ENT,       
+  KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     KC_M,     KC_COMM,  KC_DOT,   KC_SLSH,    KC_RSFT,            KC_UP,      
+  KC_LCTL,  KC_LALT,  KC_LGUI,                      KC_SPC,                       KC_RALT,  MO(Macfn),KC_APP,     KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RGHT),
+
+[Winfn] = LAYOUT(
+  KC_ESC,   KC_MYCM,  KC_MAIL,  KC_WSCH,  KC_WHOM,  KC_CALC,  KC_MSEL,  KC_MPRV,  KC_MPLY,  KC_MNXT,  KC_MUTE,    KC_VOLD,  KC_VOLU,  RGB_TOG,  KC_SCRL,  KC_PAUS,  
+  KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,       KC_MINS,  KC_EQL,   KC_BSPC,  KC_INS,   KC_HOME,  RGB_HUI,  
+  KC_TAB,   KC_Q,     TO(Win),  KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,       KC_LBRC,  KC_RBRC,  KC_BSLS,  RGB_MOD,  KC_END,   RGB_HUD,   
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,    KC_QUOT,            KC_ENT,      
+  KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     TO(Mac),  KC_COMM,  KC_DOT,   KC_SLSH,    KC_RSFT,            RGB_VAI,    
+  KC_LCTL,  GUI_TOG,  KC_LALT,                      KC_SPC,                       KC_RALT,  MO(Winfn),KC_APP,     KC_RCTL,  RGB_SPD,  RGB_VAD,  RGB_SPI),
+
+[Macfn] = LAYOUT(
+  KC_ESC,   KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,     KC_F11,   KC_F12,   RGB_TOG,  KC_SCRL,  KC_PAUS, 
+  KC_GRV,   KC_1,     KC_2,     KC_3,     KC_4,     KC_5,     KC_6,     KC_7,     KC_8,     KC_9,     KC_0,       KC_MINS,  KC_EQL,   KC_BSPC,  KC_INS,   KC_HOME,  RGB_HUI,  
+  KC_TAB,   KC_Q,     TO(Win),  KC_E,     KC_R,     KC_T,     KC_Y,     KC_U,     KC_I,     KC_O,     KC_P,       KC_LBRC,  KC_RBRC,  KC_BSLS,  RGB_MOD,  KC_END,   RGB_HUD,    
+  KC_CAPS,  KC_A,     KC_S,     KC_D,     KC_F,     KC_G,     KC_H,     KC_J,     KC_K,     KC_L,     KC_SCLN,    KC_QUOT,            KC_ENT,      
+  KC_LSFT,  KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,     KC_N,     TO(Mac),  KC_COMM,  KC_DOT,   KC_SLSH,    KC_RSFT,            RGB_VAI,     
+  KC_LCTL,  KC_LALT,  KC_LGUI,                      KC_SPC,                       KC_RALT,  MO(Winfn),KC_APP,     KC_RCTL,  RGB_SPD,  RGB_VAD,  RGB_SPI),
+
+};
+
+
+void led_init_ports(void) {
+  setPinOutput(C0);
+  setPinOutput(C14);
+  setPinOutput(C15);
+  writePinLow(C15);
+  writePinLow(C0);
+};
+
```

Comment:
This initialization needs to be moved to the keyboard level code, so that a keymap with no custom code (e.g., a configurator build, which basically contains just the `keymaps` array and nothing else) would not end up with uninitialized hardware.  You should also provide some useful behavior at the keyboard level (the Caps Lock handling is already in the core, the Win Lock indicator is somewhat more messy to implement in a way that allows overriding its functionality at the keymap level, and the Mac mode indicator would need to remain keymap-specific, but you may set it up as Scroll Lock at the keyboard level).